backend/src/api/lightning/clightning/clightning-convert.ts

```diff
@@ -71,8 +72,11 @@ export async function convertAndmergeBidirectionalChannels(clChannels: any[]): P
 }
 
 export function convertChannelId(channelId): string {
+  if (channelId.indexOf('/') !== -1) {
+    channelId = channelId.slice(0, -2);
+  }
   const s = channelId.split('x').map(part => parseInt(part));
-  return BigInt((s[0] << 40) | (s[1] << 16) | s[2]).toString();
+  return bolt07_encode_chan_id({block_height: s[0], block_index: s[1], output_index: s[2]});
```

Comment:
the value inside `BigInt()`, `(s[0] << 40) | (s[1] << 16) | s[2]` is not a bigint, it is a javascript number, which only has 52 bits. precision is lost here and the channel id will have the last few digits replaced by zeros

this can be fixed by converting each part of the id to bigint first:
```ts
const s = channelId.split('x').map(part => BigInt(part));
return ((s[0] << 40n) | (s[1] << 16n) | s[2]).toString();
```

Reply:
The code you are commenting has been deleted. Do you mean we can use the old code with this simple fix?

Reply:
yes